### DISCUSSION TITLE
Access theia behind an HTTP Proxy Server.

### DISCUSSION THREAD

**Discussion by kumbham** [eclipse-theia/theia#8807] [Q&A]
I'm accessing theia using an HTTP proxy server and I have observed that in order to access the theia using the proxy endpoint, the proxy needs to open/upgrade the HTTP connection to a WebSocket connection. I'm curious why do we need to upgrade the connection to WebSocket? If I don't upgrade the connection to a WebSocket connection, theia will display a loading spinner endlessly.

Setup:
* theia runs on a remote machine on port 3000.
* Sample http proxy server code:

```
var http = require('http'),
var httpProxy = require('http-proxy');
var proxy = httpProxy.createProxyServer({});
httpProxy.createProxyServer({ target: '<remote_ip>:3000', ws: true }).listen(8000);  // upgrades http connection to websocket connection.
```

**Comment by pawansingh00** [comment]
@kumbham With this editor was opening for you? For me it seems to be opening the editor (file explorer etc.) but file is not opening and all (Websocket connection never gets response) and `proxy.on('close'` is getting called.

Details can be seen here - https://github.com/eclipse-theia/theia/discussions/12229#top
